src/Hint/Structure.hs

```diff
@@ -39,6 +39,7 @@ foo = let ~(x:xs) = y in z
 module Hint.Structure(structureHint) where
 
 import Hint.Type
+import Data.Function (on)
```

Comment:
Implicit import lists decrease readability, is many cases you can't figure out where does a variable come from.

Implicit import lists may cause name conflicts if some library has added new name. Adding a new name is considered a non-breaking change in Hackage PVP and in Semver.

So, implicit imports are a bad style.